src/login/LoginOptionsScreen.tsx

```diff
@@ -83,11 +72,15 @@ export default function LoginOptionsScreen({
       if (!onboarded) {
         await completeOnboarding();
       }
-      navigation.dispatch(
-        StackActions.replace(afterLogin.routeName as any, {
-          ...afterLogin.routeParams,
-        }),
-      );
+      // @TODO Verify that this wont mess things up for android
+      navigation.navigate(afterLogin.routeName as any, {
+        ...afterLogin.routeParams,
+      });
+      // navigation.dispatch(
```

Comment:
The reason we had to use  **StackActions.replace** here was because we have two stacks, one with Generic onboarding + login stack, and other with the rest of the stacks. Now since after login through generic onboarding, we dont have the after login route name in the present stack, it is not able to simple navigate.
Can we go back to StackActions.replace or an alternative of that in the new version.? 



https://user-images.githubusercontent.com/29625161/188460548-6c39a18f-8194-40ac-a143-5ee49867edea.mov

Reply:
Might be that this is also causing Mobile token onboarding to not appear when needed. I see @rosvik has commented that the screens pop up twice and maybe you have added this commit to fix that. But yes that is a known case and kind of a side effect of how navigation is handled in the app currently. But we have discussed it offline here and found it acceptable 😅 . cc @gorandalum

Reply:
FYI I think also some of the issues are due to breaking changes in `.navigate`. Previousl all `navigate` calls implicitly merged params (if having existing params on that screen). Now you have to explicitly mark it with `merge: true`.

But I'll look into this more and re-add replace. Good context and thanks! This was the reason I didn't delete it as I didn't understand the consequences, so this is very valuable input.

Reply:
Hm. Changing this to replace again it loads the screen twice after completing onboarding. Is that the "acceptable" part you are talking about?

Reply:
yes 😅